### PR TITLE
Intermediate CSS & HTML - Default Styles: Rewrite lesson

### LINKED ISSUE #27126: Intermediate HTML and CSS - Default Styles: Revise lesson
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### Describe your suggestion

I would like to rewrite the Default Styles lesson significantly.  I think there is much room for improvement here, which will help learners better understand resets and their options.

Some thoughts supporting this:

1) I think links to the resets are additional resources, not assignments.  I feel assignments should be activities that actually teach the learner something rather than simply being a resource.
2) I believe, the listed resources are outdated; is Myer really the most popular?  It is out of date, heavy, and makes troubleshooting difficult.  The linked Normalize is no longer being maintained; we should link to the new one.
3) I think we should discuss the idea of teams/learners having their resets developed using these other ones as a foundation but customized for the team/developer.
4) Would it be too much to talk about opinionated vs. unopinionated?  I kinda feel this distinction may be important for evaluating resets and their contents.

I would like to research more up-to-date articles to include in the assignment section and then provide a handful of example sheets in the Additional resources.

I would also like to include a link to a tool I made for visualizing resets: https://fabulousgk.github.io/fabulous-styleboard/

### Path

Ruby / Rails, Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/intermediate-html-and-css-default-styles

### (Optional) Discord Name

_No response_

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by fabulousgk**
## Because

The current version of the lesson lacks in many ways.  It contains links to resets as asignments, contains outdaated information, and does not do a great job of explaing the topic.

## This PR

- Rewrites the lesson

## Issue

Closes #27126

## Additional Information

This is a big rewrite so feedback is appreciated.  As to the linting error, not much I can do there, the correct name is Normalize.css.

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [ x The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: intermediate_html_css/intermediate_css_concepts/default_styles.md
Comment (nikitarevenco) [inline intermediate_html_css/intermediate_css_concepts/default_styles.md:27]: Love his stuff. His articles are just fire

**Review by MaoShizhong — COMMENTED** [review]
Absolute :fire: rewrite!
I agree with a lot of the main ideas driving the rewrite, as well as the changes to structure/framing as a whole.

Just a couple of wording suggestions and nit fixes below from me.

**Comment by fabulousgk** [comment]
Thank you @MaoShizhong I accepted your changes, they all were great!

**Comment by MaoShizhong** [comment]
Sorry it's been taking a while - just been looking for a second opinion before merging given it's a lesson rewrite (even if a little smaller than something like Factory Functions).

**Comment by fabulousgk** [comment]
@MaoShizhong TY!

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Mix of change requests and general comments below. Overall this is looking like a great improvement, though 💪🏼

File: intermediate_html_css/intermediate_css_concepts/default_styles.md
Comment (fabulousgk) [reply intermediate_html_css/intermediate_css_concepts/default_styles.md:12]: I have some heading changes, that will be in my next commit...

File: intermediate_html_css/intermediate_css_concepts/default_styles.md
Comment (fabulousgk) [reply intermediate_html_css/intermediate_css_concepts/default_styles.md:28]: I think this was for the styleboard link...it is showing wierd to me.  If so, I have not problem moving it.  My intent was simply to have learners see visually how these sheets change things in different ways.  But the additional resources is fine.

File: intermediate_html_css/intermediate_css_concepts/default_styles.md
Comment (fabulousgk) [reply intermediate_html_css/intermediate_css_concepts/default_styles.md:27]: Makes sense to me, but I am conflicted as to which one to remove.  I am leaning towards Elad's as it is a bit advanced, but I like that it is such a dramatically different approach.  A second for me woudl be Coumeau's, but I love his stuff, so I just cant decide...

I could use some opions on this....

File: intermediate_html_css/intermediate_css_concepts/default_styles.md
Comment (fabulousgk) [reply intermediate_html_css/intermediate_css_concepts/default_styles.md:44]: I like them as it gives you a reference, whcih I think is important personaly.  But I am not particullary attached to them.

Perhaps something like "You may want to bookmark these for reference"?

File: intermediate_html_css/intermediate_css_concepts/default_styles.md
Comment (fabulousgk) [reply intermediate_html_css/intermediate_css_concepts/default_styles.md:45]: I am quite alright with axing them.  I kept them as I did not want to offend anyone, but I agree with you.

Comment (fabulousgk) [reply intermediate_html_css/intermediate_css_concepts/default_styles.md:45]: I also think that allowing someone to search these out is positive as they then have to do a bit more reading to understand them.

**Comment by fabulousgk** [comment]
@thatblindgeye I pushed a new commit with some changes, and will await comment on the others to work on that.

File: intermediate_html_css/intermediate_css_concepts/default_styles.md
Comment (thatblindgeye) [reply intermediate_html_css/intermediate_css_concepts/default_styles.md:31]: We can make @MaoShizhong decide 😈 

My first thought was Elad's as well for the same reasons you stated, but honestly it might be between that and the "Making the case for CSS normalize and reset stylesheets in 2023" one if we wanted to cut at least one.

Again, I'd also be good with shipping all of them and let feedback help us determine if we need to cut some stuff out as a followup.

Comment (MaoShizhong) [reply intermediate_html_css/intermediate_css_concepts/default_styles.md:31]: Let me read between the Elad and CSS Normalize ones and see how I feel whether we should definitely keep both or definitely axe one.

Comment (MaoShizhong) [reply intermediate_html_css/intermediate_css_concepts/default_styles.md:31]: A more thorough read-through of all of them, I'm of the opinion that we can drop the Elad article.
They're all great reads, don't get me wrong, and the Elad one definitely has some benefits from going deeper with a much more different approach from some of the others, but in the context of someone at this point in the curriculum, I don't feel like that more advanced and different angle will be as beneficial/easy to appreciate as it may be to people like us with a little more practical understand of this all.

I think the other 3 articles are sufficient to get the main points across while reining in the scope to a more sensible radius. How does that sound?

Comment (fabulousgk) [reply intermediate_html_css/intermediate_css_concepts/default_styles.md:31]: Ok, done.

**Comment by fabulousgk** [comment]
@thatblindgeye and @MaoShizhong  I added a couple new videos to the lineup based on this: https://github.com/TheOdinProject/curriculum/issues/27779

**Comment by fabulousgk** [comment]
Uhh the linter has gone crazy

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Just a few more comments below. As for the linter, it's something with GitHub's beta check annotations running the linter on unchanged files. Don't think there's a way to turn that off via a setting (which is confusing why they wouldn't allow that if so...), but maybe something that can be tweaked with the workflow yml file. Nothing to worry about here, though.

File: intermediate_html_css/intermediate_css_concepts/default_styles.md
Comment (fabulousgk) [reply intermediate_html_css/intermediate_css_concepts/default_styles.md:48]: Ok, removed

File: intermediate_html_css/intermediate_css_concepts/default_styles.md
Comment (fabulousgk) [reply intermediate_html_css/intermediate_css_concepts/default_styles.md:46]: ok

File: intermediate_html_css/intermediate_css_concepts/default_styles.md
Comment (MaoShizhong) [reply intermediate_html_css/intermediate_css_concepts/default_styles.md:31]: Just watched it. In the current site's lesson, I think it'd be a good addition. But now that a lot has been rewritten with different framing of some parts so it's less "these are CSS resets. Use them" and more "this is what a CSS reset means. Here are some ideas some people like about that, but not everyone uses them or the same ones", I think this video is not needed. It covers a lot of what has already been said and besides some small details like "some resets just do things for IE", still just is a guy giving his own opinion on how he does his reset (a super minimal but still using some rules-reset).

**Review by thatblindgeye — APPROVED** [review]

**Comment by MaoShizhong** [comment]
@fabulousgk How come this was closed?

**Comment by fabulousgk** [comment]
OMG. I saw the approval from thatblindgeye and assumed.  Uhhh, ohh look there is a restore branch button, I hope it works.

**Comment by fabulousgk** [comment]
Thank god

**Comment by fabulousgk** [comment]
@MaoShizhong I cant reopen, but I was able to restore the branch

**Comment by MaoShizhong** [comment]
Panic XD He was just waiting for me to do a once over and merge but I was out and just got back. Glad I came back in time!

**Review by MaoShizhong — APPROVED** [review]
We are good to go! :rocket:

**Comment by fabulousgk** [comment]
Thank you, I need to pay better attention to the emails and what they are actually saying :D

**Comment by MaoShizhong** [comment]
Now at least we know there's a quick and easy way to restore a deleted branch in these cases!

**Pull request merged**
